test/integration/model.tests.js

```diff
@@ -795,6 +802,137 @@ describe('Model', function () {
       });
     });
 
+    it('should create a record in the database that will properly expire with given ttl', function (done) {
+      Foo.create({ entity: { fooId: three }, ttl: 3 }, function (err) {
+        assume(err).is.falsey();
+
+        Foo.findOne({ fooId: three }, function (error, res) {
+          assume(error).is.falsey();
+          assume(res);
+          assume(res.fooId).equals(three);
+
+          setTimeout(function () {
+            Foo.findOne({ fooId: three }, function (er, result) {
+              assume(er).is.falsey();
+              assume(result).is.falsey();
+              done();
+            });
+          }, 3000);
+        });
+      });
+    });
+
+    it('should create a record in the database that will expire but still be found before it reaches given ttl', function (done) {
+      Foo.create({ entity: { fooId: four }, ttl: 7 }, function (err) {
+        assume(err).is.falsey();
+
+        Foo.findOne({ fooId: four }, function (err, res) {
+          assume(err).is.falsey();
+          assume(res);
+          assume(res.fooId).equals(four);
+
+          setTimeout(function () {
+            Foo.findOne({ fooId: four }, function (err, result) {
+              assume(err).is.falsey();
+              assume(result);
+              assume(result.fooId).equals(four);
+              done();
+            });
+          }, 3000);
+        });
+      });
+    });
+
+    it('should update a record in the database that will expire with given ttl', function (done) {
+      Foo.update({ entity: { fooId: five, something: 'foo' }, ttl: 2 }, function (err) {
+        assume(err).is.falsey();
+
+        Foo.findOne({ fooId: five }, function (error, result) {
+          assume(error).is.falsey();
+          assume(result);
+          assume(result.fooId).equals(five);
+
+          setTimeout(function () {
+            Foo.findOne({ fooId: five }, function (er, res) {
+              assume(er).is.falsey();
+              assume(res).is.falsey();
+              done();
+            });
+          }, 5000);
```

Comment:
This can be shorter since the above ttl is 2 seconds

Reply:
updated